components/gitpod-db/go/oidc_client_config.go

```diff
@@ -29,7 +30,31 @@ func (c *OIDCClientConfig) TableName() string {
 	return "d_b_oidc_client_config"
 }
 
+// It feels wrong to have to define re-define all of these fields.
+// However, I could not find a Go library which would include json annotations on the structs to guarantee the fields
+// will remain consistent over time (and resilient to rename). If we find one, we can change this.
 type OIDCSpec struct {
+	OAuth2   OAuth2Config   `json:"oauth2"`
```

Comment:
@easyCZ can we iterate on that a little bit more? 
'Was struggling to stab this in another context here: https://github.com/gitpod-io/gitpod/commit/e2b13bcbe1d8e73a23018a692f50de01649f5f4f

Not being able to reuse existing structures is one problem, extensibility might be a bigger one.

Reply:
Just saw your comment @AlexTugarev and added a hold.
Feel free to release once you're done here. :lotus_position:

Reply:
I'm not sure I understand. What would you like to see here added or removed?

Of course this can change, it just exists as a small form which allows the current static google config to be stored, and hence remove the need for the static file.